frontend/public/components/utils/button-bar.jsx

```diff
@@ -21,12 +21,12 @@ const InfoMessage = ({message}) => <div className="alert alert-info"><span class
 // NOTE: DO NOT use <a> elements within a ButtonBar.
 // They don't support the disabled attribute, and therefore
 // can't be disabled during a pending promise/request.
-export const ButtonBar = ({children, className, errorMessage, infoMessage, inProgress}) => {
-  return <div className={classNames(className, 'co-m-btn-bar')}>
-    {errorMessage && <ErrorMessage message={errorMessage} />}
-    {injectDisabled(children, inProgress)}
-    {inProgress && <LoadingInline />}
-    {infoMessage && <InfoMessage message={infoMessage} />}
+export const ButtonBar = (props) => {
+  return <div className={classNames(props.className, 'co-m-btn-bar')}>
+    {props.errorMessage && <ErrorMessage message={props.errorMessage} />}
+    {injectDisabled(props.children, props.inProgress)}
+    {props.inProgress && <LoadingInline />}
+    {props.infoMessage && <InfoMessage message={props.infoMessage} />}
```

Comment:
It's not clear to me why this change was needed?

Reply:
There was an issue with the required properties for the ButtonBar
>TS2322: Type '{ children: Element[]; errorMessage: any; inProgress: boolean; }' is not assignable to type 'IntrinsicAttributes & { children: any; className: any; errorMessage: any; infoMessage: any; inPro...'.
  Type '{ children: Element[]; errorMessage: any; inProgress: boolean; }' is not assignable to type '{ children: any; className: any; errorMessage: any; infoMessage: any; inProgress: any; }'.
    Property 'className' is missing in type '{ children: Element[]; errorMessage: any; inProgress: boolean; }'.
Version: typescript 2.7.2
Time: 12738ms

It was cause by the listed params in the class definition
> export const ButtonBar = ({children, className, errorMessage, infoMessage, inProgress}) => {
...
}

Because of listing the parameters and not just setting the to `props`, all the listed params will be required, even if `propTypes` will declare that they are not.